services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/modification/AbstractModifySubscriptionHandler.java

```diff
@@ -185,4 +205,94 @@ protected abstract String[] getRequiredPermissions(ModifySubscriptionRequest req
     protected void validateAdditionalPermissionRules(ModifySubscriptionRequest request,
             @Nullable ContextInfo contextInfo) {}
 
+    /**
+     * Generates the name for a cart based on the operationFlow triggered and subscription.
+     * 
+     * @param operationFlowName the flow under execution currently
+     * @param subscription the current subscription
+     * @return a unique cart name String
+     * @see DefaultSubscriptionActionFlow
+     */
+    protected String generateCartName(@NonNull String operationFlowName,
+            @NonNull Subscription subscription) {
+        return "%s %s - %s".formatted(operationFlowName, subscription.getName(),
+                ULID.random(SECURE_RANDOM));
+    }
+
+    /**
+     * Build the {@link PriceCartRequest} for pricing the cart.
+     * 
+     * @param subscription the current subscription
+     * @param contextInfo the context info
+     * @return the generated PriceCartRequest
+     */
+    protected PriceCartRequest buildPriceCartRequest(@lombok.NonNull Subscription subscription,
+            @Nullable ContextInfo contextInfo) {
+        PriceCartRequest priceCartRequest = typeFactory.get(PriceCartRequest.class);
+        priceCartRequest.setCurrency(subscription.getCurrency());
+        return priceCartRequest;
+    }
+
+    protected void hydrateRequest(@NonNull ModifySubscriptionRequest request,
+            @Nullable ContextInfo contextInfo) {
+        setExistingAndNewProductsOnModifySubscriptionRequest(request, contextInfo);
+    }
+
+    protected void setExistingAndNewProductsOnModifySubscriptionRequest(
+            @NonNull ModifySubscriptionRequest request,
+            @Nullable ContextInfo contextInfo) {}
+
+    protected void setExistingAndNewProductsOnModifySubscriptionRequest(
+            @NonNull ModifySubscriptionRequest request,
+            @NonNull CatalogProvider<Product> catalogProvider,
+            @Nullable ContextInfo contextInfo) {
+        Pair<Product, Product> newAndRootItems =
+                readNewAndRootItems(request, catalogProvider, contextInfo);
+
+        request.setProductToChangeTo(newAndRootItems.getLeft());
+        request.setRootProduct(newAndRootItems.getRight());
+    }
+
+    protected Pair<Product, Product> readNewAndRootItems(
+            @NonNull ModifySubscriptionRequest request,
+            @NonNull CatalogProvider<Product> catalogProvider,
+            @Nullable ContextInfo contextInfo) {
```

Comment:
Better to be more generic to support future needs that may require more products to be read. The caller can figure out which Product is which since they know what they were asking for.
```suggestion
    protected <T extends Product> List<T> readNewAndRootItems(
            @NonNull ModifySubscriptionRequest request,
            @NonNull CatalogProvider<T> catalogProvider,
            @Nullable ContextInfo contextInfo) {
```

Reply:
My comment is moot with the changes I made in this PR https://github.com/BroadleafCommerce/SubscriptionOperationServices/pull/20.